### PR TITLE
Pr/db setup

### PR DESCRIPTION
**Pull request by mwestphall**
Baseline implementation, implements the following endpoints with API-key based authentication:
- `GET /documents`
- `GET /documents/{document_id}`
- `GET /documents/{document_id}/content`
- `GET /documents/{document_id}/page/{page_num}`
- `GET /documents/{document_id}/page/{page_num}/snippet/{snippet_bounding_box}`

Caching of pages/snippets is implemented via an S3 temporary object policy. This can be set up manually via the s3 API, but we'll probably want to automate bucket configuration at some point

```python
s3.put_bucket_lifecycle_configuration(Bucket='public-pdfs',LifecycleConfiguration={
    'Rules': [
        {"Filter":{"Prefix":"pages/"},"Status":"Enabled","Expiration":{"Days":1}},
        {"Filter":{"Prefix":"snippets/"},"Status":"Enabled","Expiration":{"Days":1}},
    ]
})
```

### CODE REVIEW

File: src/app.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+@prefix_router.get("/documents/{document_id}/content")
+def get_document_contents(document_id: UUID, x_api_key: Optional[str] = Header(None)) -> RedirectResponse:
+    """ Return a redirect to the full PDF contents of the given document """
+    article = get_article_with_auth(document_id, x_api_key)
+    operator = PdfOperator(article)
+    return RedirectResponse(operator.get_presigned_document())
+
+@prefix_router.get("/documents/{document_id}/page/{page_num}")
+def get_document_snippet(
```
Comment (iross) [inline src/app.py]: Suggest: rename to `get_document_page`

File: src/app.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
+from os import environ
+from fastapi import FastAPI, APIRouter, Header
+from fastapi.responses import RedirectResponse
+from fastapi.middleware.gzip import GZipMiddleware
+from fastapi.logger import logger
```
Comment (iross) [inline src/app.py]: Unused import

File: src/db/db_models.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    pass
+
+class DbArticle(Base):
+    """ ORM Mapping for articles in the document store """
+    __tablename__ = "articles"
+
+    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid4)
+    title = Column(String)
+    xdd_doc_id = Column(String, unique=True, default=None)
+    doi = Column(String, unique=True, default=None)
```
Comment (iross) [inline src/db/db_models.py:16]: Sanity check: Does this allow for multiple PDFs without DOIs being added, or does the unique constraint prohibit it? (and similar for xdd docid in the line above).

File: src/s3/s3_client.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
+from os import environ
+from boto3 import Session
+
+session = Session(aws_access_key_id=environ['S3_ACCESS_KEY'], aws_secret_access_key=environ['S3_SECRET_KEY'])
+s3 = session.client("s3", endpoint_url="https://s3.xdd-pdfstore.chtc.io", region_name='us-east-1')
```
Comment (iross) [inline src/s3/s3_client.py]: Suggestion: set `endpoint_url` via env var (or similar)

**Review by iross — COMMENTED** [review]
Looks great! Just a few questions and minor comments added.

Does the PDF ingestion piece make sense to add here as well?

**Comment by iross** [comment]
It would also be nice to have a simple notebook showing basic usage, similar to what we did for the COSMOS service.

File: src/db/db_models.py (hunk 3 above)
Comment (mwestphall) [reply src/db/db_models.py:16]: Had to double check on this, but the default behavior of postgres is to treat nulls as distinct from other nulls. So you can have any number of nulls in a unique nullable column.

**Comment by mwestphall** [comment]
> Looks great! Just a few questions and minor comments added.
> 
> Does the PDF ingestion piece make sense to add here as well?

Yeah, I think it makes sense to add a separate Dockerfile to build a second container with a different entrypoint for doing the ingest

**Review by iross — APPROVED** [review]
👍

**Pull request merged**